### PR TITLE
Clean up and refactor the code

### PR DESCRIPTION
**Pull request by Aursen**

### CODE REVIEW

File: cron/src/time_unit/mod.rs [hunk 1]
```diff
@@ -163,7 +163,7 @@ where
         //println!("ordinals_from_specifier for {} => {:?}", Self::name(), specifier);
         match *specifier {
             All => Ok(Self::supported_ordinals().clone()),
-            Point(ordinal) => Ok((&[ordinal]).iter().cloned().collect()),
+            Point(ordinal) => Ok(OrdinalSet::from([ordinal])),
```
Comment (Adrena-Corto) [inline cron/src/time_unit/mod.rs:166]: 👍

File: plugin/src/builders/thread_exec.rs [hunk 2]
```diff
@@ -222,16 +222,6 @@ fn build_kickoff_ix(
     // Build the instruction.
     let mut kickoff_ix = match thread {
         VersionedThread::V1(_) => Instruction {
```
Comment (Adrena-Corto) [inline plugin/src/builders/thread_exec.rs:224]: so we remove v1 and rename v2 to v1?

File: programs/network/src/constants.rs [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+pub const SEED_FEE: &[u8] = b"fee";
+
+#[constant]
+pub const SEED_EPOCH: &[u8] = b"epoch";
+
+#[constant]
+pub const SEED_PENALTY: &[u8] = b"penalty";
+
+#[constant]
+pub const SEED_SNAPSHOT_ENTRY: &[u8] = b"snapshot_entry";
```
Comment (Adrena-Corto) [inline programs/network/src/constants.rs:19]: 👍

File: programs/network/src/instructions/delegation_claim.rs [hunk 4]
```diff
@@ -28,7 +30,7 @@ pub fn handler(ctx: Context<DelegationClaim>, amount: u64) -> Result<()> {
     let delegation = &mut ctx.accounts.delegation;
 
     // Decrement the delegation's claimable balance.
-    delegation.yield_balance = delegation.yield_balance.checked_sub(amount).unwrap();
+    delegation.yield_balance -= amount;
```
Comment (Adrena-Corto) [inline programs/network/src/instructions/delegation_claim.rs:33]: quid of underflow check here, is that checked elsewhere? (is that even important/relevant in that context)

File: programs/network/src/instructions/delegation_create.rs [hunk 5]
```diff
@@ -46,15 +39,6 @@ pub struct DelegationCreate<'info> {
...
     pub mint: Account<'info, Mint>,
 
-    #[account(address = sysvar::rent::ID)]
-    pub rent: Sysvar<'info, Rent>,
-
-    #[account(address = system_program::ID)]
-    pub system_program: Program<'info, System>,
-
-    #[account(address = anchor_spl::token::ID)]
-    pub token_program: Program<'info, Token>,
```
Comment (Adrena-Corto) [inline programs/network/src/instructions/delegation_create.rs:56]: Ancient Solana™️

File: programs/network/src/instructions/delegation_create.rs [hunk 6]
```diff
@@ -76,7 +66,7 @@ pub fn handler(ctx: Context<DelegationCreate>) -> Result<()> {
     delegation.init(authority.key(), worker.total_delegations, worker.key())?;
 
     // Increment the worker's total delegations counter.
-    worker.total_delegations = worker.total_delegations.checked_add(1).unwrap();
+    worker.total_delegations += 1;
```
Comment (Adrena-Corto) [inline programs/network/src/instructions/delegation_create.rs:69]: u64 prob won't oveflow but I usually write checked math for all ops like a nerd

File: programs/network/src/instructions/fee_collect.rs [hunk 7]
```diff
@@ -26,15 +25,15 @@ pub fn handler(ctx: Context<FeeCollect>, amount: u64, penalty: bool) -> Result<(
 
     // Increment the collected fee counter.
     if penalty {
-        fee.penalty_balance = fee.penalty_balance.checked_add(amount).unwrap();
+        fee.penalty_balance += amount;
```
Comment (Adrena-Corto) [inline programs/network/src/instructions/fee_collect.rs:28]: This checked arithmetic one seems important

File: programs/network/src/instructions/fee_collect.rs [hunk 8]
```diff
@@ -45,12 +44,7 @@ pub fn handler(ctx: Context<FeeCollect>, amount: u64, penalty: bool) -> Result<(
     );
 
     require!(
-        fee.collected_balance
-            .checked_add(fee.penalty_balance)
-            .unwrap()
-            .checked_add(min_rent_balance)
-            .unwrap()
-            .ge(&lamport_balance),
+        (fee.collected_balance + fee.penalty_balance + min_rent_balance) >= lamport_balance
```
Comment (Adrena-Corto) [inline programs/network/src/instructions/fee_collect.rs:47]: same here

File: programs/network/src/instructions/delegation_claim.rs (hunk 4 above)
Comment (Aursen) [reply programs/network/src/instructions/delegation_claim.rs:33]: We lost a lot of performance using this profile. But we can simplify the syntaxe, the overflow will cause panic
```toml
[profile.release]
overflow-checks = true
```

File: plugin/src/builders/thread_exec.rs (hunk 2 above)
Comment (Aursen) [reply plugin/src/builders/thread_exec.rs:224]: Yes, that's the goal, I plan to reset versioning to 1.0 when it's ready

File: programs/network/src/instructions/delegation_claim.rs (hunk 4 above)
Comment (Adrena-Corto) [reply programs/network/src/instructions/delegation_claim.rs:33]: Oh yhea right. Performance as in "at runtime" or for compilation

Comment (Aursen) [reply programs/network/src/instructions/delegation_claim.rs:33]: runtime

**Pull request merged**